source/Calamari/Kubernetes/ResourceStatus/Timer.cs

```diff
@@ -9,43 +10,23 @@ namespace Calamari.Kubernetes.ResourceStatus
     /// </summary>
     public interface ITimer
     {
+        TimeSpan Interval { get; set; }
+        TimeSpan Duration { get; set; }
         void Start();
+        void Restart();
         bool HasCompleted();
         void WaitForInterval();
     }
-    
-    /// <summary>
-    /// <inheritdoc />
-    /// </summary>
+
     public class Timer : ITimer
     {
-        private readonly TimeSpan duration;
-        private readonly TimeSpan interval;
-        private readonly Stopwatch stopwatch;
-        
-        public Timer(TimeSpan duration, TimeSpan interval)
-        {
-            this.duration = duration;
-            this.interval = interval;
-            stopwatch = new Stopwatch();
-        }
-
+        private readonly Stopwatch stopwatch = new Stopwatch();
+        public TimeSpan Interval { get; set; } = TimeSpan.FromSeconds(1);
+        public TimeSpan Duration { get; set; } = Timeout.InfiniteTimeSpan;
```

Comment:
If possible, I still think it is better to have these in the constructor to enforce consistency. The default values are a little bit unobvious, eg. we'll never actually use the default interval I think. Good work on `Timeout.InfiniteTimeSpan` though, I didn't know about it.

Reply:
Good point - I've updated this to use Delegate Factories and I think it looks pretty neat.

Reply:
I also just realised that the `WaitForInterval` method is using `Thread.Sleep`. I've updated it to use `Task.Delay` (as an async method) because it is generally advised to avoid `Thread.Sleep` and we're in an async context when we call the method anyway.